### PR TITLE
Include Collection objects in collection bags, and allow single collection objects to be bagged without children

### PR DESCRIPTION
**Pull request by bondjimbond**
**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-2402, https://jira.duraspace.org/browse/ISLANDORA-1305)

# What does this Pull Request do?

Provides an additional option for creating Collection bags: Collection only, no children. Produces a Bag of collection metadata and other datastreams. Also includes the Collection object in Collection bags.

# What's new?

Now offers the ability to treat a Collection as an object like any other. If the 'no_children' option is selected, when you choose "create bag" for a collection object, you'll get a nice little bag of the Collection's rels-ext, mods, etc.

This is useful when you want a Collection bag for future preservation purposes, and you've got bags of child objects already and don't want to re-bag them.

If creating a collection bag with its child objects, includes the collection object itself as part of the bag.

# How should this be tested?

- Check out this branch
- Choose the new option in the Bagit config
- Create a collection bag; you should get one, without any children attached.
- Change options and create another collection bag, see that collection object is now included in the generated bag

# Interested parties
@mjordan  @Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by bondjimbond** [comment]
Updated this to fully address ISLANDORA-1305. Would appreciate a review from @mjordan; once merged this will impact how islandora_westvault will need to work.

**Comment by bondjimbond** [comment]
Discussed at last Committers' Call; agreed this is needed. Needs reviewer.

**Comment by mjordan** [comment]
I will review.

**Review by mjordan — APPROVED** [review]
Works as advertised.

**Comment by bondjimbond** [comment]
@mjordan Ready to merge? Once this is in, I can merge https://github.com/bondjimbond/islandora_westvault/pull/31

**Comment by bondjimbond** [comment]
Thanks! (Also, I cited the wrong pull in the above comment, but all's well.)

**Pull request merged**
